tempodb/encoding/vparquet2/block_traceql.go

```diff
@@ -1874,3 +1868,25 @@ func newSpanAttr(name string) traceql.Attribute {
 func newResAttr(name string) traceql.Attribute {
 	return traceql.NewScopedAttribute(traceql.AttributeScopeResource, false, name)
 }
+
+func unionIfNeeded(definitionLevel int, iters []parquetquery.Iterator, pred parquetquery.GroupPredicate) parquetquery.Iterator {
+	switch len(iters) {
```

Comment:
are there savings here? one less set of `parquetquery.RowNumber`s to increment and track? if so i'm a bit surprised we're not seeing better #s on benchmarks.

in my profiling we spend a lot of time testing and incrementing row numbers so i would have expected better returns here. can we do something similar with the `LeftJoinIterator`?

Reply:
Right, less looping/peeking/copying of the intermediate results.  Savings depend on the query.  The only benchmark that triggers this is the new `{} | count() > 1` which is how I caught it but haven't measured the amount of improvement due to `unionIfNeeded` versus the batchCollector changes. 

It's hard to reason about but `{ span.http.status_code=500} | count() > 2` also triggers this and shows a nice improvement:

```
name                          old time/op    new time/op    delta
BackendBlockTraceQL/5xxcount    77.5ms ± 2%    70.2ms ± 1%   -9.40%  (p=0.000 n=10+9)

name                          old speed      new speed      delta
BackendBlockTraceQL/5xxcount   451MB/s ± 2%   498MB/s ± 1%  +10.36%  (p=0.000 n=10+9)

name                          old MB_io/op   new MB_io/op   delta
BackendBlockTraceQL/5xxcount      34.9 ± 0%      34.9 ± 0%     ~     (all equal)

name                          old alloc/op   new alloc/op   delta
BackendBlockTraceQL/5xxcount    1.78MB ± 0%    1.79MB ± 0%   +0.33%  (p=0.000 n=9+10)

name                          old allocs/op  new allocs/op  delta
BackendBlockTraceQL/5xxcount     22.8k ± 0%     22.7k ± 0%   -0.23%  (p=0.000 n=7+10)
```